spec/console_spec.rb

```diff
@@ -0,0 +1,175 @@
+      allow(subject.instance_variable_set(:@respondent, respondent_double))
+      expect(subject).to receive(:new_process)
+      subject.choose_action
+    end
+
+    it 'when guessed code - win and agree to save result' do
+      allow(subject).to receive(:input).and_return('start' , '1234', 'y')
+      allow(subject).to receive(:game_state_valid?).and_return(true, false)
+      allow(subject).to receive(:process_helper).and_return(process_helper_double)
+      # allow(subject.instance_variable_set(:@instance_process_helper, process_helper_double))
+      allow(process_helper_double).to receive(:setup_player)
+      allow(process_helper_double).to receive(:setup_difficulty)
+      allow(subject).to receive(:game).and_return(game_double)
+      # allow(subject.instance_variable_set(:@instance_game, game_double))
+      allow(game_double).to receive(:attempt).and_return(:guessed_code)
+      allow(game_double).to receive(:errors).and_return([])
+      allow(subject).to receive(:set_game_options)
+      allow(game_double).to receive(:winner).and_return(true)
+      expect(subject).to receive(:save_to_db)
+      expect(subject).to receive(:new_process)
+      subject.choose_action
+    end
+
+    it 'when guessed code - win and do not save result' do
+      allow(subject).to receive(:input).and_return('start' , '1234', 'y')
+      allow(subject).to receive(:game_state_valid?).and_return(true, false)
+      allow(subject).to receive(:process_helper).and_return(process_helper_double)
+      # allow(subject.instance_variable_set(:@instance_process_helper, process_helper_double))
+      allow(process_helper_double).to receive(:setup_player)
+      allow(process_helper_double).to receive(:setup_difficulty)
+      allow(subject).to receive(:game).and_return(game_double)
+      # allow(subject.instance_variable_set(:@instance_game, game_double))
+      allow(game_double).to receive(:attempt).and_return(:guessed_code)
+      allow(game_double).to receive(:errors).and_return([])
+      allow(subject).to receive(:set_game_options)
+      allow(game_double).to receive(:winner).and_return(true)
+      expect(subject).to receive(:new_process)
+      subject.choose_action
+    end
+
+    it 'with statistics output' do
+      allow(subject).to receive(:input).and_return('stats', 'start')
+      allow(respondent_double).to receive(:show_message)
+      allow(respondent_double).to receive(:show)
+      allow(statistics_double).to receive(:winners)
+      allow(subject).to receive(:statistic).and_return(statistics_double)
+      # allow(subject.instance_variable_set(:@instance_statistic, statistics_double))
+      allow(subject).to receive(:respondent).and_return(respondent_double)
+      expect(subject).to receive(:process)
+      # expect(respondent).to receive(:show)
+      # expect(subject).to receive(:respondent).to receive(:show)
+      subject.choose_action
+    end
+  end
+end
+
+RSpec.describe Statistics do
```

Comment:
why do you spec Statistics module here?
Read about how to spec Modules

https://stackoverflow.com/questions/1542945/testing-modules-in-rspec